app/models/tipline_newsletter.rb

```diff
@@ -134,6 +134,7 @@ def whatsapp_template_name
   end
 
   def articles
+    return @articles unless @articles.to_a.empty?
```

Comment:
Avoid re-fetching articles from RSS feed if this same method is called many times for the same object already in memory. It's like a cheap cache.